cmd/busyambassador/main.go

```diff
@@ -6,110 +6,62 @@
 package main
 
 import (
-	"bufio"
+	// stdlib
 	"context"
 	"fmt"
 	"os"
 	"strings"
 
+	// 3rd-party libs
+	"github.com/kballard/go-shellquote"
+
+	// 1st-party libs
 	"github.com/datawire/ambassador/v2/pkg/busy"
 	"github.com/datawire/ambassador/v2/pkg/environment"
 
-	amb_agent "github.com/datawire/ambassador/v2/cmd/agent"
+	// commands
+	"github.com/datawire/ambassador/v2/cmd/agent"
 	"github.com/datawire/ambassador/v2/cmd/ambex"
 	"github.com/datawire/ambassador/v2/cmd/apiext"
 	"github.com/datawire/ambassador/v2/cmd/entrypoint"
 	"github.com/datawire/ambassador/v2/cmd/kubestatus"
 	"github.com/datawire/ambassador/v2/cmd/reproducer"
 )
 
-// Version is inserted at build-time using --ldflags -X
-var Version = "(unknown version)"
-
 func noop(_ context.Context) {}
 
 // Builtin for showing this image's version.
-func showVersion(ctx context.Context, Version string, args ...string) error {
-	fmt.Printf("Version %s\n", Version)
+func showVersion(ctx context.Context, version string, args ...string) error {
+	fmt.Printf("Version %s\n", version)
 
 	return nil
 }
 
 func main() {
-	// Allow ambassador.version to override the compiled-in Version.
-	//
-	// "Wait wait wait wait wait," I hear you cry. "Why in the world are you
-	// doing this??" Two reasons:
-	//
-	// 1. ambassador.version is updated during the RC and GA process to always
-	//    contain the Most Polite Version of the version number -- this is the
-	//    ONE thing that should be shown to users.
-	// 2. We do _not_ recompile busyambassador during the RC and GA process, and
-	//    we don't want to: we want to ship the bits we tested, and while we're
-	//    OK with altering a text file after that, recompiling feels weirder. So
-	//    we don't.
+	// The version number is set at run-time by reading the 'ambassador.version' file.  We do
+	// this instead of compiling in a version so that we can promote RC images to GA without
+	// recompiling anything.
 	//
-	// End result: fall back on the compiled-in version, but let ambassador.version
-	// be the primary.
-
-	// THIS IS A CLOSURE CALL, not just an anonymous function definition. Making the
-	// call lets me defer file.Close().
-	func() {
-		file, err := os.Open("/buildroot/ambassador/python/ambassador.version")
-
-		if err != nil {
-			// We DON'T log errors here; we just silently fall back to the
-			// compiled-in version. This is because the code in main() here is
-			// running _wicked_ early, and logging setup happens _after_ this
-			// function.
-			//
-			// XXX Letting the logging setup happen here, instead, would likely
-			// be an improvement.
-			return
-		}
-
-		defer file.Close()
-
-		// Read line by line and hunt for BUILD_VERSION.
-		scanner := bufio.NewScanner(file)
-
-		for scanner.Scan() {
-			line := scanner.Text()
-
+	// Keep this parsing logic in-sync with VERSION.py.
+	version := "dirty"
+	if verBytes, err := os.ReadFile("/buildroot/ambassador/python/ambassador.version"); err == nil {
```

Comment:
This needs a comment. We _check_ for `err == nil`, but if `err` is not `nil` we don't log or anything, we just silently set `version` to `dirty`. The old version talked at length about why we don't log, and (obviously) I thought that was important.